### PR TITLE
Print full ReFrame test support in bot report

### PR DESCRIPTION
**Pull request by casparvl**

### CODE REVIEW

**Comment by casparvl** [comment]
bot: build repo:eessi.io-2023.06-software arch:zen2

**Comment by casparvl** [comment]
bot: build repo:eessi.io-2023.06-software arch:zen2

**Comment by casparvl** [comment]
bot: build repo:eessi.io-2023.06-software arch:zen2

**Comment by casparvl** [comment]
bot: build repo:eessi.io-2023.06-software arch:zen2

**Comment by casparvl** [comment]
bot: build repo:eessi.io-2023.06-software arch:zen2

**Comment by casparvl** [comment]
bot: build repo:eessi.io-2023.06-software arch:zen2

**Comment by casparvl** [comment]
Hm, it's not reporting back. I think last time that happened, the payload of the message was too large. In this case, the number of characters for the test step is aroud 6500. Note that this is a _big_ downside to printing the full test results. Long term, maybe we should look at different ways to do this (a gist?).

But... I don't understand why I don't get a report back. 6500 should be _well_ below the character limit for github comments, which I believe is around 65k.

**Comment by boegel** [comment]
> Hm, it's not reporting back. I think last time that happened, the payload of the message was too large. In this case, the number of characters for the test step is aroud 6500. Note that this is a _big_ downside to printing the full test results. Long term, maybe we should look at different ways to do this (a gist?).
> 
> But... I don't understand why I don't get a report back. 6500 should be _well_ below the character limit for github comments, which I believe is around 65k.

See crash output below, we'll need to escape `%` it seems...
<details>

```
Traceback (most recent call last):
  File "/usr/lib64/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/usr/lib64/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/home/bot/eessi-bot-software-layer/eessi_bot_job_manager.py", line 733, in <module>
    main()
  File "/home/bot/eessi-bot-software-layer/eessi_bot_job_manager.py", line 724, in main
    job_manager.process_finished_job(known_jobs[fj])
  File "/home/bot/eessi-bot-software-layer/eessi_bot_job_manager.py", line 538, in process_finished_job
    comment_description = job_tests.get(job_metadata.JOB_TEST_COMMENT_DESCRIPTION, comment_description)
  File "/usr/lib64/python3.6/configparser.py", line 1283, in get
    fallback=fallback, **kwargs)
  File "/usr/lib64/python3.6/configparser.py", line 800, in get
    d)
  File "/usr/lib64/python3.6/configparser.py", line 394, in before_get
    self._interpolate_some(parser, option, L, value, section, defaults, 1)
  File "/usr/lib64/python3.6/configparser.py", line 444, in _interpolate_some
    "found: %r" % (rest,))
configparser.InterpolationSyntaxError: '%' must be followed by '%' or '(', found: "%scale=1_node %device_type=cpu %module_name=LAMMPS/29Aug2024-foss-2023b-kokkos /aeb2d9df @BotBuildTests:x86-64-amd-zen2-node+default<br/>\x1b[33mWARNING: hooks.req_memory_per_node does not support the scheduler you configured (local). The test will run, but since it doesn't request the required amount of memory explicitely, it may result in an out-of-memory error. Please expand the functionality of hooks.req_memory_per_node for your scheduler.\x1b[0m<br/>[ RUN      ] EESSI_LAMMPS_lj %scale=1_node %device_type=cpu %module_name=LAMMPS/2Aug2023_update2-foss-2023a-kokkos /04ff9ece @BotBuildTests:x86-64-amd-zen2-node+default<br/>\x1b[33mWARNING: hooks.req_memory_per_node does not support the scheduler you configured (local). The test will run, but since it doesn't request the required amount of memory explicitely, it may result in an out-of-memory error. Please expand the functionality of hooks.req_memory_per_node for your scheduler.\x1b[0m<br/>[ RUN      ] EESSI_OSU_Micro_Benchmarks_coll %benchmark_info=mpi.collective.osu_allreduce %scale=1_node %module_name=OSU-Micro-Benchmarks/7.2-gompi-2023b %device_type=cpu /31ac6ab9 @BotBuildTests:x86-64-amd-zen2-node+default<br/>[ RUN      ] EESSI_OSU_Micro_Benchmarks_coll %benchmark_info=mpi.collective.osu_allreduce %scale=1_node %module_name=OSU-Micro-Benchmarks/7.1-1-gompi-2023a %device_type=cpu /f3be40a2 @BotBuildTests:x86-64-amd-zen2-node+default<br/>[ RUN      ] EESSI_OSU_Micro_Benchmarks_coll %benchmark_info=mpi.collective.osu_alltoall %scale=1_node %module_name=OSU-Micro-Benchmarks/7.2-gompi-2023b %device_type=cpu /10e66fba @BotBuildTests:x86-64-amd-zen2-node+default<br/>[ RUN      ] EESSI_OSU_Micro_Benchmarks_coll %benchmark_info=mpi.collective.osu_alltoall %scale=1_node %module_name=OSU-Micro-Benchmarks/7.1-1-gompi-2023a %device_type=cpu /5be57ae7 @BotBuildTests:x86-64-amd-zen2-node+default<br/>[ RUN      ] EESSI_OSU_Micro_Benchmarks_pt2pt %benchmark_info=mpi.pt2pt.osu_latency %scale=1_node %module_name=OSU-Micro-Benchmarks/7.2-gompi-2023b %device_type=cpu /c8c9aff5 @BotBuildTests:x86-64-amd-zen2-node+default<br/>[ RUN      ] EESSI_OSU_Micro_Benchmarks_pt2pt %benchmark_info=mpi.pt2pt.osu_latency %scale=1_node %module_name=OSU-Micro-Benchmarks/7.1-1-gompi-2023a %device_type=cpu /9795e491 @BotBuildTests:x86-64-amd-zen2-node+default<br/>[ RUN      ] EESSI_OSU_Micro_Benchmarks_pt2pt %benchmark_info=mpi.pt2pt.osu_bw %scale=1_node %module_name=OSU-Micro-Benchmarks/7.2-gompi-2023b %device_type=cpu /48da21c5 @BotBuildTests:x86-64-amd-zen2-node+default<br/>[ RUN      ] EESSI_OSU_Micro_Benchmarks_pt2pt %benchmark_info=mpi.pt2pt.osu_bw %scale=1_node %module_name=OSU-Micro-Benchmarks/7.1-1-gompi-2023a %device_type=cpu /1b8c1ca2 @BotBuildTests:x86-64-amd-zen2-node+default<br/>[       OK ] ( 1/10) EESSI_LAMMPS_lj %scale=1_node %device_type=cpu %module_name=LAMMPS/29Aug2024-foss-2023b-kokkos /aeb2d9df @BotBuildTests:x86-64-amd-zen2-node+default<br/>P: perf: 441.04 timesteps/s (r:0, l:None, u:None)<br/>[       OK ] ( 2/10) EESSI_LAMMPS_lj %scale=1_node %device_type=cpu %module_name=LAMMPS/2Aug2023_update2-foss-2023a-kokkos /04ff9ece @BotBuildTests:x86-64-amd-zen2-node+default<br/>P: perf: 447.937 timesteps/s (r:0, l:None, u:None)<br/>[       OK ] ( 3/10) EESSI_OSU_Micro_Benchmarks_coll %benchmark_info=mpi.collective.osu_allreduce %scale=1_node %module_name=OSU-Micro-Benchmarks/7.2-gompi-2023b %device_type=cpu /31ac6ab9 @BotBuildTests:x86-64-amd-zen2-node+default<br/>P: latency: 4.65 us (r:0, l:None, u:None)<br/>[       OK ] ( 4/10) EESSI_OSU_Micro_Benchmarks_coll %benchmark_info=mpi.collective.osu_allreduce %scale=1_node %module_name=OSU-Micro-Benchmarks/7.1-1-gompi-2023a %device_type=cpu /f3be40a2 @BotBuildTests:x86-64-amd-zen2-node+default<br/>P: latency: 4.74 us (r:0, l:None, u:None)<br/>[       OK ] ( 5/10) EESSI_OSU_Micro_Benchmarks_coll %benchmark_info=mpi.collective.osu_alltoall %scale=1_node %module_name=OSU-Micro-Benchmarks/7.2-gompi-2023b %device_type=cpu /10e66fba @BotBuildTests:x86-64-amd-zen2-node+default<br/>P: latency: 8.84 us (r:0, l:None, u:None)<br/>[       OK ] ( 6/10) EESSI_OSU_Micro_Benchmarks_coll %benchmark_info=mpi.collective.osu_alltoall %scale=1_node %module_name=OSU-Micro-Benchmarks/7.1-1-gompi-2023a %device_type=cpu /5be57ae7 @BotBuildTests:x86-64-amd-zen2-node+default<br/>P: latency: 8.55 us (r:0, l:None, u:None)<br/>[       OK ] ( 7/10) EESSI_OSU_Micro_Benchmarks_pt2pt %benchmark_info=mpi.pt2pt.osu_latency %scale=1_node %module_name=OSU-Micro-Benchmarks/7.2-gompi-2023b %device_type=cpu /c8c9aff5 @BotBuildTests:x86-64-amd-zen2-node+default<br/>P: latency: 0.34 us (r:0, l:None, u:None)<br/>[       OK ] ( 8/10) EESSI_OSU_Micro_Benchmarks_pt2pt %benchmark_info=mpi.pt2pt.osu_latency %scale=1_node %module_name=OSU-Micro-Benchmarks/7.1-1-gompi-2023a %device_type=cpu /9795e491 @BotBuildTests:x86-64-amd-zen2-node+default<br/>P: latency: 0.34 us (r:0, l:None, u:None)<br/>[       OK ] ( 9/10) EESSI_OSU_Micro_Benchmarks_pt2pt %benchmark_info=mpi.pt2pt.osu_bw %scale=1_node %module_name=OSU-Micro-Benchmarks/7.2-gompi-2023b %device_type=cpu /48da21c5 @BotBuildTests:x86-64-amd-zen2-node+default<br/>P: bandwidth: 7559.9 MB/s (r:0, l:None, u:None)<br/>[       OK ] (10/10) EESSI_OSU_Micro_Benchmarks_pt2pt %benchmark_info=mpi.pt2pt.osu_bw %scale=1_node %module_name=OSU-Micro-Benchmarks/7.1-1-gompi-2023a %device_type=cpu /1b8c1ca2 @BotBuildTests:x86-64-amd-zen2-node+default<br/>P: bandwidth: 7574.35 MB/s (r:0, l:None, u:None)<br/>[----------] all spawned checks have finished<br/><br/>[  PASSED  ] Ran 10/10 test case(s) from 10 check(s) (0 failure(s), 0 skipped, 0 aborted)<br/>[==========] Finished on Fri Oct  4 21<br/></dd><dt>_Details_</dt><dd>:white_check_mark: job output file <code>slurm-21389.out</code><br/>:white_check_mark: no message matching <code>ERROR: </code><br/>:white_check_mark: no message matching <code>\\[\\s\\*FAILED\\s\\*\\].\\*Ran .\\* test case</code><br/></dd></dl></details>"
```

</details>

**Comment by casparvl** [comment]
bot: build repo:eessi.io-2023.06-software arch:zen2

**Comment by casparvl** [comment]
bot: build repo:eessi.io-2023.06-software arch:zen2

**Comment by casparvl** [comment]
I don't know why, I still have unescaped `%` signs. I had thought the `sed` command would replace those. I guess I need to try harder and try interactively... :)

**Comment by casparvl** [comment]
bot: build repo:eessi.io-2023.06-software arch:zen2

**Comment by casparvl** [comment]
bot: build repo:eessi.io-2023.06-software arch:zen2

**Comment by casparvl** [comment]
This looks good in terms of `%` chars. Is there more that I should escape? Or is `\`not the right escape character?

```
[TEST]
comment_description = <details><summary>:grin: SUCCESS _(click triangle for details)_</summary><dl><dt>_ReFrame Summary_</dt><dd>[----------] start processing checks<br/>[ RUN      ] EESSI_LAMMPS_lj \%scale=1_node \%device_type=cpu \%module_name=LAMMPS/29Aug2024-foss-2023b-kokkos /aeb2d9df @BotBuildTests:x86-64-amd-zen2-node+default<br/>ESC[33mWARNING: hooks.req_memory_per_node does not support the scheduler you configured (local). The test will run, but since it doesn't request the required amount of memory explicitely, it may result in an out-of-memory error. Please expand the functionality of hooks.req_memory_per_node for your scheduler.ESC[0m<br/>[ RUN      ] EESSI_LAMMPS_lj \%scale=1_node \%device_type=cpu \%module_name=LAMMPS/2Aug2023_update2-foss-2023a-kokkos /04ff9ece @BotBuildTests:x86-64-amd-zen2-node+default<br/>ESC[33mWARNING: hooks.req_memory_per_node does not support the scheduler you configured (local). The test will run, but since it doesn't request the required amount of memory explicitely, it may result in an out-of-memory error. Please expand the functionality of hooks.req_memory_per_node for your scheduler.ESC[0m<br/>[ RUN      ] EESSI_OSU_Micro_Benchmarks_coll \%benchmark_info=mpi.collective.osu_allreduce \%scale=1_node \%module_name=OSU-Micro-Benchmarks/7.2-gompi-2023b \%device_type=cpu /31ac6ab9 @BotBuildTests:x86-64-amd-zen2-node+default<br/>[ RUN      ] EESSI_OSU_Micro_Benchmarks_coll \%benchmark_info=mpi.collective.osu_allreduce \%scale=1_node \%module_name=OSU-Micro-Benchmarks/7.1-1-gompi-2023a \%device_type=cpu /f3be40a2 @BotBuildTests:x86-64-amd-zen2-node+default<br/>[ RUN      ] EESSI_OSU_Micro_Benchmarks_coll \%benchmark_info=mpi.collective.osu_alltoall \%scale=1_node \%module_name=OSU-Micro-Benchmarks/7.2-gompi-2023b \%device_type=cpu /10e66fba @BotBuildTests:x86-64-amd-zen2-node+default<br/>[ RUN      ] EESSI_OSU_Micro_Benchmarks_coll \%benchmark_info=mpi.collective.osu_alltoall \%scale=1_node \%module_name=OSU-Micro-Benchmarks/7.1-1-gompi-2023a \%device_type=cpu /5be57ae7 @BotBuildTests:x86-64-amd-zen2-node+default<br/>[ RUN      ] EESSI_OSU_Micro_Benchmarks_pt2pt \%benchmark_info=mpi.pt2pt.osu_latency \%scale=1_node \%module_name=OSU-Micro-Benchmarks/7.2-gompi-2023b \%device_type=cpu /c8c9aff5 @BotBuildTests:x86-64-amd-zen2-node+default<br/>[ RUN      ] EESSI_OSU_Micro_Benchmarks_pt2pt \%benchmark_info=mpi.pt2pt.osu_latency \%scale=1_node \%module_name=OSU-Micro-Benchmarks/7.1-1-gompi-2023a \%device_type=cpu /9795e491 @BotBuildTests:x86-64-amd-zen2-node+default<br/>[ RUN      ] EESSI_OSU_Micro_Benchmarks_pt2pt \%benchmark_info=mpi.pt2pt.osu_bw \%scale=1_node \%module_name=OSU-Micro-Benchmarks/7.2-gompi-2023b \%device_type=cpu /48da21c5 @BotBuildTests:x86-64-amd-zen2-node+default<br/>[ RUN      ] EESSI_OSU_Micro_Benchmarks_pt2pt \%benchmark_info=mpi.pt2pt.osu_bw \%scale=1_node \%module_name=OSU-Micro-Benchmarks/7.1-1-gompi-2023a \%device_type=cpu /1b8c1ca2 @BotBuildTests:x86-64-amd-zen2-node+default<br/>[       OK ] ( 1/10) EESSI_LAMMPS_lj \%scale=1_node \%device_type=cpu \%module_name=LAMMPS/29Aug2024-foss-2023b-kokkos /aeb2d9df @BotBuildTests:x86-64-amd-zen2-node+default<br/>P: perf: 440.947 timesteps/s (r:0, l:None, u:None)<br/>[       OK ] ( 2/10) EESSI_LAMMPS_lj \%scale=1_node \%device_type=cpu \%module_name=LAMMPS/2Aug2023_update2-foss-2023a-kokkos /04ff9ece @BotBuildTests:x86-64-amd-zen2-node+default<br/>P: perf: 441.529 timesteps/s (r:0, l:None, u:None)<br/>[       OK ] ( 3/10) EESSI_OSU_Micro_Benchmarks_coll \%benchmark_info=mpi.collective.osu_allreduce \%scale=1_node \%module_name=OSU-Micro-Benchmarks/7.2-gompi-2023b \%device_type=cpu /31ac6ab9 @BotBuildTests:x86-64-amd-zen2-node+default<br/>P: latency: 4.19 us (r:0, l:None, u:None)<br/>[       OK ] ( 4/10) EESSI_OSU_Micro_Benchmarks_coll \%benchmark_info=mpi.collective.osu_allreduce \%scale=1_node \%module_name=OSU-Micro-Benchmarks/7.1-1-gompi-2023a \%device_type=cpu /f3be40a2 @BotBuildTests:x86-64-amd-zen2-node+default<br/>P: latency: 4.75 us (r:0, l:None, u:None)<br/>[       OK ] ( 5/10) EESSI_OSU_Micro_Benchmarks_coll \%benchmark_info=mpi.collective.osu_alltoall \%scale=1_node \%module_name=OSU-Micro-Benchmarks/7.2-gompi-2023b \%device_type=cpu /10e66fba @BotBuildTests:x86-64-amd-zen2-node+default<br/>P: latency: 8.72 us (r:0, l:None, u:None)<br/>[       OK ] ( 6/10) EESSI_OSU_Micro_Benchmarks_coll \%benchmark_info=mpi.collective.osu_alltoall \%scale=1_node \%module_name=OSU-Micro-Benchmarks/7.1-1-gompi-2023a \%device_type=cpu /5be57ae7 @BotBuildTests:x86-64-amd-zen2-node+default<br/>P: latency: 8.57 us (r:0, l:None, u:None)<br/>[       OK ] ( 7/10) EESSI_OSU_Micro_Benchmarks_pt2pt \%benchmark_info=mpi.pt2pt.osu_latency \%scale=1_node \%module_name=OSU-Micro-Benchmarks/7.2-gompi-2023b \%device_type=cpu /c8c9aff5 @BotBuildTests:x86-64-amd-zen2-node+default<br/>P: latency: 0.32 us (r:0, l:None, u:None)<br/>[       OK ] ( 8/10) EESSI_OSU_Micro_Benchmarks_pt2pt \%benchmark_info=mpi.pt2pt.osu_latency \%scale=1_node \%module_name=OSU-Micro-Benchmarks/7.1-1-gompi-2023a \%device_type=cpu /9795e491 @BotBuildTests:x86-64-amd-zen2-node+default<br/>P: latency: 0.32 us (r:0, l:None, u:None)<br/>[       OK ] ( 9/10) EESSI_OSU_Micro_Benchmarks_pt2pt \%benchmark_info=mpi.pt2pt.osu_bw \%scale=1_node \%module_name=OSU-Micro-Benchmarks/7.2-gompi-2023b \%device_type=cpu /48da21c5 @BotBuildTests:x86-64-amd-zen2-node+default<br/>P: bandwidth: 7686.87 MB/s (r:0, l:None, u:None)<br/>[       OK ] (10/10) EESSI_OSU_Micro_Benchmarks_pt2pt \%benchmark_info=mpi.pt2pt.osu_bw \%scale=1_node \%module_name=OSU-Micro-Benchmarks/7.1-1-gompi-2023a \%device_type=cpu /1b8c1ca2 @BotBuildTests:x86-64-amd-zen2-node+default<br/>P: bandwidth: 7739.59 MB/s (r:0, l:None, u:None)<br/>[----------] all spawned checks have finished<br/><br/>[  PASSED  ] Ran 10/10 test case(s) from 10 check(s) (0 failure(s), 0 skipped, 0 aborted)<br/>[==========] Finished on Mon Oct  7 20<br/></dd><dt>_Details_</dt><dd>:white_check_mark: job output file <code>slurm-21924.out</code><br/>:white_check_mark: no message matching <code>ERROR: </code><br/>:white_check_mark: no message matching <code>\[\s\*FAILED\s\*\].\*Ran .\* test case</code><br/></dd></dl></details>
```

**Comment by casparvl** [comment]
bot: build repo:eessi.io-2023.06-software arch:zen2

**Comment by casparvl** [comment]
Ok, well that's a good step. It's a lot of output though, and it quickly could bring us to the comment character limit (65k or so). For reference, the above ReFrame output is about 6.5k chars.

I'll try to strip the
```
[ RUN ] EESSI_OSU_Micro_Benchmarks_coll %benchmark_info=mpi.collective.osu_allreduce %scale=1_node %module_name=OSU-Micro-Benchmarks/7.2-gompi-2023b %device_type=cpu /31ac6ab9 @BotBuildTests:x86-64-amd-zen2-node+default
```
Lines from it. The only disadvantage is that if generating the test instance creates any warning, like 
```
�[33mWARNING: hooks.req_memory_per_node does not support the scheduler you configured (local). The test will run, but since it doesn't request the required amount of memory explicitely, it may result in an out-of-memory error. Please expand the functionality of hooks.req_memory_per_node for your scheduler.�[0m
```
You cannot see which test instantiation caused this warning to be printed.

Alternatively, I could grep on the pattern

```
[ OK ] (10/10) EESSI_OSU_Micro_Benchmarks_pt2pt %benchmark_info=mpi.pt2pt.osu_bw %scale=1_node %module_name=OSU-Micro-Benchmarks/7.1-1-gompi-2023a %device_type=cpu /1b8c1ca2 @BotBuildTests:x86-64-amd-zen2-node+default
``` 

with something like `\[.*\]\S*\([0-9]+/[0-9]+\).*<br\/>` or something. (i.e. try to capture the `[ OK ] (10/10)` and then anything until end of line, which in this message now contains <br\/>). Or, if I do it before the replacement with <br\/>, it'd be `$` I guess.

Then, I can grep for a second pattern that matches
```
[ PASSED ] Ran 10/10 test case(s) from 10 check(s) (0 failure(s), 0 skipped, 0 aborted)
```
and simply append that. It would make the message twice as short, and not include any warnings from the test instantiation (then again, those _could_ be important and a reason for further investigation at least...)

**Comment by casparvl** [comment]
bot: build repo:eessi.io-2023.06-software arch:zen2

**Comment by casparvl** [comment]
bot: build repo:eessi.io-2023.06-software arch:zen2

**Comment by casparvl** [comment]
bot: build repo:eessi.io-2023.06-software arch:zen2

**Comment by casparvl** [comment]
bot: build repo:eessi.io-2023.06-software arch:zen2

**Comment by laraPPr** [comment]
@casparvl it seems like the configs on the build cluster need to be updated?
Is that for another pr or are we leaving it like this?
```
WARNING: hooks.req_memory_per_node does not support the scheduler you configured (local). The test will run, but since it doesn't request the required amount of memory explicitely, it may result in an out-of-memory error. Please expand the functionality of hooks.req_memory_per_node for your scheduler.
```

**Comment by laraPPr** [comment]
bot: build repo:eessi.io-2023.06-software arch:zen4

**Review by laraPPr — APPROVED** [review]
lgtm

**Comment by laraPPr** [comment]
I think we should first maybe resolve the `Warning` in the output?

File: bot/check-test.sh [hunk 1]
```diff
@@ -77,9 +77,38 @@ if [[ ! -z ${grep_reframe_failed} ]]; then
...
+        # Remove the line '[==========] Finished on Mon Oct  7 21'
+        grep -v '\[=*\]' | \
+        # Remove blank line(s) from the report
+        grep -v '^$' | \
+        # Strip color coding characters
+        sed 's/\x1B\[[0-9;]*m//g' | \
+        # Replace all newline characters with <br/>
+        sed ':a;N;$!ba;s/\n/<br\/>/g' | \
+        # Replace % with %%. Use \%\% to interpret both %% as (non-special) characters
+        sed 's/\%/\%\%/g' \
```
Comment (laraPPr) [inline bot/check-test.sh:101]: skip `WARNING: hooks.req_memory_per_node` lines

**Review by laraPPr — CHANGES_REQUESTED** [review]

**Comment by casparvl** [comment]
bot: build repo:eessi.io-2023.06-software arch:zen2

**Pull request merged**